RAGchain/benchmark/dataset/dstc.py

```diff
@@ -0,0 +1,168 @@
+
+        # Create gt_passages for ingest.
+        gt_passages = ingest_data[ingest_data['doc_id'].isin(id_for_remove_duplicated_corpus)]
+        gt_passages = gt_passages.apply(self.__make_passages, axis=1).tolist()
+
+        if ingest_size is not None:
+            # ingest size must be larger than evaluate size.
+            if ingest_size >= self.eval_size:
+                ingest_data = ingest_data.sample(n=ingest_size, replace=False, random_state=random_state,
+                                                 axis=0)
+            else:
+                raise ValueError("ingest size must be same or larger than evaluate size")
+
+        # Remove duplicated passages between corpus and retrieval gt for ingesting passages faster.
+        # Marking duplicated values in the corpus using retrieval_gt id.
+        mask = ingest_data.isin(id_for_remove_duplicated_corpus)
+        # Remove duplicated passages
+        ingest_data = ingest_data[~mask.any(axis=1)]
+        passages = ingest_data.apply(self.__make_passages, axis=1).tolist()
+        passages += gt_passages
+
+        for retrieval in retrievals:
+            retrieval.ingest(passages)
+        db.create_or_load()
+        db.save(passages)
+
+    def evaluate(self, **kwargs) -> EvaluateResult:
+
+        return self._calculate_metrics(
+            questions=list(self.questions),
+            pipeline=self.run_pipeline,
+            retrieval_gt=list(self.retrieval_gt),
+            answer_gt=list([answer] for answer in self.response),
+            **kwargs
+        )
+
+    def __make_passages(self, row):
+        if row['doc_type'] == 'review':
+            content = row['review_sentence']
+            metadata_etc = {
+                'domain': row['domain'],
+                'entity_id': row['entity_id'],
+                'entity_name': row['entity_name'],
+                'doc_type': row['doc_type'],
+                **row['review_metadata']
+            }
+
+        elif row['doc_type'] == 'faq':
+            content = row['faq_question'] + ', ' + row['faq_answer']
+            metadata_etc = {
+                'domain': row['domain'],
+                'entity_id': row['entity_id'],
+                'entity_name': row['entity_name'],
+                'doc_type': row['doc_type'],
+            }
+
+        return Passage(
+            id=row['doc_id'],
+            content=content,
+            filepath=self.file_path,
+            metadata_etc=metadata_etc
+        )
+
+    def __renewal_doc_id(self, row):
+        if row['doc_type'] == 'review':
+            return str(row['doc_id']) + '_' + row['doc_type'] + '_' + row['domain'] \
+                + '_' + str(row['entity_id']) + '_' + str(row['review_sent_id'])
+        elif row['doc_type'] == 'faq':
+            return str(row['doc_id']) + '_' + row['doc_type'] + '_' \
+                + row['domain'] + '_' + str(row['entity_id'])
+
+    def __preprocess_qa(self, row):
+        question = row['log'][-1]['text']
```

Comment:
use whole conversation instead of last question.
DSTC11-Track5 have conversation history, so better use whole conversation and put it into question. 

There are no clear way to directly input whole chat history to pipeline. I know.
As alternative, make conversation history to prompt (string type) and insert to `question`